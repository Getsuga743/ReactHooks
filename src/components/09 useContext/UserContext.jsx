/* eslint-disable import/prefer-default-export */
import { createContext } from 'react';

const UserContext = createContext(null);

export default UserContext;
