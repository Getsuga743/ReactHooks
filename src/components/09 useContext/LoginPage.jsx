import React, { useContext } from 'react';
import UserContext from './UserContext';

const LoginPage = () => {
  const { setUser } = useContext(UserContext);
  return (
    <div className="container-fluid">
      <h1>LoginPage</h1>
      <button
        className="btn btn-primary"
        onClick={() => {
          setUser({ id: 123, name: 'fernando' });
        }}
        type="button"
      >
        Login
      </button>
    </div>
  );
};

export default LoginPage;
