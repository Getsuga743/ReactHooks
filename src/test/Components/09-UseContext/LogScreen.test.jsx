/* eslint-disable no-unused-vars */
/* eslint-disable no-undef */
import React, { useContext } from 'react';
import { mount } from 'enzyme';
import LoginPage from '../../../components/09 useContext/LoginPage';
import UserContext from '../../../components/09 useContext/UserContext';
import '../../../setupTest';

describe('Pruebas en <LoginPage />', () => {
  const setUser = jest.fn();
  const wrapper = mount(
    // eslint-disable-next-line react/jsx-no-comment-textnodes
    <UserContext.Provider value={{ setUser }}>
      // eslint-disable-next-line react/jsx-no-undef
      <LoginPage />
    </UserContext.Provider>,
  );
  test('debe de mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('debe de ejecutar el setUser con el argumento esperado', () => {
    wrapper.find('button').prop('onClick')();
    expect(setUser).toHaveBeenCalledWith({
      id: 123,
      name: 'fernando',
    });
  });
});
