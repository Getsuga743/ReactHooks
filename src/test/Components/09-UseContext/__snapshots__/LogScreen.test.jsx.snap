// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <LoginPage /> debe de mostrarse correctamente 1`] = `
Array [
  "// eslint-disable-next-line react/jsx-no-undef",
  <div
    className="container-fluid"
  >
    <h1>
      LoginPage
    </h1>
    <button
      className="btn btn-primary"
      onClick={[Function]}
      type="button"
    >
      Login
    </button>
  </div>,
]
`;
